lib/marked.js

```diff
@@ -1186,125 +1186,130 @@ Parser.prototype.parseText = function() {
+        while (this.next().type !== 'list_end') {
+          body += this.tok();
+        }
 
-      if (this.token.task) {
-        if (loose) {
-          if (this.peek().type === 'text') {
-            var nextToken = this.peek();
-            nextToken.text = this.renderer.checkbox(checked) + ' ' + nextToken.text;
+        return this.renderer.list(body, ordered, start);
+      }
+      case 'list_item_start': {
+        body = '';
+        var loose = this.token.loose;
+        var checked = this.token.checked;
+        var task = this.token.task;
+
+        if (this.token.task) {
+          if (loose) {
+            if (this.peek().type === 'text') {
+              var nextToken = this.peek();
+              nextToken.text = this.renderer.checkbox(checked) + ' ' + nextToken.text;
+            } else {
+              this.tokens.push({
+                type: 'text',
+                text: this.renderer.checkbox(checked)
+              });
+            }
           } else {
-            this.tokens.push({
-              type: 'text',
-              text: this.renderer.checkbox(checked)
-            });
+            body += this.renderer.checkbox(checked);
           }
-        } else {
-          body += this.renderer.checkbox(checked);
         }
-      }
 
-      while (this.next().type !== 'list_item_end') {
-        body += !loose && this.token.type === 'text'
-          ? this.parseText()
-          : this.tok();
+        while (this.next().type !== 'list_item_end') {
+          body += !loose && this.token.type === 'text'
+            ? this.parseText()
+            : this.tok();
+        }
+        return this.renderer.listitem(body, task, checked);
       }
-      return this.renderer.listitem(body, task, checked);
-    }
-    case 'html': {
-      // TODO parse inline content if parameter markdown=1
-      return this.renderer.html(this.token.text);
-    }
-    case 'paragraph': {
-      return this.renderer.paragraph(this.inline.output(this.token.text));
-    }
-    case 'text': {
-      return this.renderer.paragraph(this.parseText());
-    }
-    default: {
-      var errMsg = 'Token with "' + this.token.type + '" type was not found.';
-      if (this.options.silent) {
-        console.log(errMsg);
-      } else {
-        throw new Error(errMsg);
+      case 'html': {
+        // TODO parse inline content if parameter markdown=1
+        return this.renderer.html(this.token.text);
+      }
+      case 'paragraph': {
+        return this.renderer.paragraph(this.inline.output(this.token.text));
+      }
+      case 'text': {
+        return this.renderer.paragraph(this.parseText());
+      }
+      default: {
+        var errMsg = 'Token with "' + this.token.type + '" type was not found.';
+        if (this.options.silent) {
+          console.log(errMsg);
+        } else {
+          throw new Error(errMsg);
+        }
       }
     }
+  } catch (err) {
+    console.error(err) // let developer see stacktrace
```

Comment:
I feel like we still need to throw the same error so users can log it how they want. We could just change the `err.message` and rethrow it like on [line 1651](https://github.com/markedjs/marked/blob/master/lib/marked.js#L1651)